torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3129,6 +3129,33 @@ def _maybe_guard_eq(self, expr: Union["sympy.Eq", "sympy.Ne"], concrete_bool: bo
                 r = try_solve(expr, mod_expr, floordiv_inequality=False)
                 if r is not None and r[1] == 0:
                     self.divisible.add(mod_expr)
+                    # This is a little bit of extra logic to make things like
+                    # torch.empty(i0, q).view(c, -1, q) work out
+                    p, q = mod_expr.args
+                    if isinstance(q, sympy.Number) and isinstance(p, sympy.Mul) and len(p.args) == 2:
+                        c, i0 = p.args
+                        # Given Mod(c * i0, q) == 0
+                        if (
+                            isinstance(c, sympy.Number) and
+                            isinstance(i0, sympy.Symbol) and
+                            self.is_unbacked_symint(i0) and
+                            int(q) % int(c) == 0
```

Comment:
As discussed, you can get rid of this condition replacing `c` by `gcd(q, c)` as the smallest positive solution of `Mod(c * i0, q) == 0` is `i0 = q/gcd(q, c)`